collectors/EsxiOnlineStateCollector.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from os import getenv
+import datetime
+from prometheus_client.core import GaugeMetricFamily
+
+import modules.TimedBlacklist as blacklist
+from BaseCollector import BaseCollector
+
+# init logging
+logger = logging.getLogger('esxi-exporter')
+
+
+class EsxiOnlineStateCollector(BaseCollector):
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def collect(self):
+
+        starttime = datetime.datetime.now()
+
+        gauge_metric = GaugeMetricFamily('esxi_online_state', '1=online, 0=offline',
+                                         labels=['vcenter', 'host', 'origin'])
+
+        # get all hosts from vcenter
+        vc_hosts = self.get_vcenter_hosts()
+        for host in vc_hosts:
+            vc_state = host.overallStatus == 'green'
```

Comment:
This overall status is pretty useful. Put it in a seperate collector and monitor all states.

Reply:
Done.